fossas/fossa-cli
# Fix Conflicting Types for "revision" in fossa-yml schema

# Overview

I've been spending a lot of time in the .fossa.yml file lately, and the erroneous red squiggly has become increasingly bothersome.

This change prevents the "Expected Array" error message from appearing when using the `revision` key.

When pasting our [example](https://github.com/fossas/fossa-cli/blob/master/docs/references/files/fossa-yml.md) .fossa.yml deps file, the following occurs:
![image](https://github.com/fossas/fossa-cli/assets/47155269/dc1f18a4-6079-4871-aea7-a79d7f929c8e)

Ideally, we'd have no errors, as this is valid and accepted by fossa-cli.

When attempting to use an array, the CLI returns the following error:
```
[ERROR] ----------
  An issue occurred

  >>> Relevant errors

    Error

      Error parsing file: /path/to/.fossa.yml.

          Aeson exception:
          Error in $.revision: parsing ConfigRevision failed, expected Object, but encountered Array
```

After the change:
![image](https://github.com/fossas/fossa-cli/assets/47155269/b41caa83-bc2f-4b7d-9d23-9f5602458ffd)

## Acceptance criteria

Using the revision key does not throw an error, everything else works as intended.

## Testing plan

1. Checkout `master`
2. Point your VSC/IDE yaml schemas to the appropriate file.
    - (This can be done by adding the following to your settings.json)
```json
        "yaml.schemas": {
            "file:/path/to/fossa-cli/docs/references/files/fossa-deps.schema.json": "fossa-deps.yml",
            "file:/path/to/fossa-cli/docs/references/files/fossa-yml.v3.schema.json": ".fossa.yml"
        },
```
3. Create a .fossa.yml with the following:
```yaml
version: 3

revision:
  commit: "12345"
  branch: master
```
4. Observe the error
5. Checkout this branch
6. Reload VSC
7. Error resolved!

## Risks

I think this change is fine because the type is declared in [another location](https://github.com/fossas/fossa-cli/blob/master/docs/references/files/fossa-yml.v3.schema.json#L104-L106), but I don't know if the best practice is to change the type at this (PR) location to `object` or to remove it completely. Still, I guess that's what code reviews are for!

## Checklist
